### ISSUE TITLE
Inconsistent behavior of Dgram Socket.Disconnect

### ISSUE THREAD

**Issue by sakno** [dotnet/runtime#34473]
`Socket.Disconnect(boolean)` method called on UDP socket behaves differently on different platforms. On Linux (Ubuntu 18.04.03 LTS) x64 it doesn't throw exception. On Windows 10 I get the following exception:
```
Message: 
    System.Net.Sockets.SocketException : A request to send or receive data was disallowed because the socket is not connected and (when sending on a datagram socket using a sendto call) no address was supplied.
  Stack Trace: 
    Socket.UpdateStatusAfterSocketErrorAndThrowException(SocketError error, String callerName)
    Socket.Disconnect(Boolean reuseSocket)
```
Probably, behavior on Windows 10 is correct because UDP is connectionless protocol.

Version is .NET Core 3.1.201.

**Comment by sakno** [comment]
However, `Socket.Connect` works fine. It's strange because I can call `Connect` but unable to call `Disconnect`.

**Comment by karelz** [comment]
Triage: It would be nice to have unified behavior across OS's. However, there may be difference in behavior of OS as #35048 demonstrates.
If it is possible and easy, we would take a fix.

@sakno is it something you would be interested in contributing?

**Comment by sakno** [comment]
@karelz , probably yes, I can try. But I have 3 open PRs, it's not good to work in parallel at everything :smile:

**Comment by wfurt** [comment]
triage: we should investigate if we _can_ make it consistent. We can possibly throw `PlatformNotSupportedException` and we can document the behavior. Currently low priority -> leaving in Future.
